text/0374-jsii-modular.md

```diff
@@ -0,0 +1,256 @@
+# Modular jsii
+
+* **Original Author(s)**: @RomainMuller
+* **Tracking Issue**: #374
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+This RFC proposes to unlock (de-couple) the versions of all jsii packages
+(i.e: compiler, code generators, runtimes, tools and libraries) in order to
+enable introduction of new major versions of any of these packages without
+necessarily incurring a new major version of all other packages.
+
+In order to make it easier to operate with de-coupled package versions, this RFC
+further proposes we break the [aws/jsii](https://github.com/aws/jsii) mono-repo
+into multiple single-package repositories.
+
+## Working Backwards
+
+### Release Notes entry
+
+```md
+Future releases of the *jsii* package constellation will no longer follow an
+all-at-once version increment strategy. Instead, each individual package will be
+released under their own version line. It is hence no longer necessary to ensure
+the version of all *jsii* packages in a product's dependency are aligned on the
+exact same release. For example, it is now possible to have a dependency on the
+`jsii` compiler at `4.3.0` while using `jsii-pacmak` at `1.35.0`, which
+previously would have led to a fatal error when executing `jsii-pacmak`.
```

Comment:
There must be some version order requirement between the producer and consumer sides, no?

Specifically when the feature expects properties produced by the compiler.